src/vault.cairo

```diff
@@ -65,6 +68,25 @@ mod Vault {
         amount: TokenAmount,
     }
 
+    #[derive(Drop, starknet::Event)]
+    struct ContractAdded {
```

Comment:
Please, add a `token` field so it's easier to distinguish what type of vault it is.